src/db/DBOperations.ts

```diff
@@ -0,0 +1,10 @@
+import { FhirResourceType } from '@projecttacoma/node-fhir-server-core';
+import { Connection } from './Connection';
+
+/**
+ * searches the database for the desired resource and returns the data
+ */
+export const findResourceById = async (id: string, resourceType: FhirResourceType) => {
+  const collection = Connection.db.collection(resourceType);
+  return collection.findOne({ id: id }, { projection: { _id: 0 } }) as any;
+};
```

Comment:
For now I think we can just use the generic on the invocation of `findOne` to get type safety instead. Also, let's change this to an `export async function` instead of export `const`:


``` ts
export async function findResourceById<T extends fhir4.FhirResource>(id: string, resourceType: FhirResourceType) {
  const collection = Connection.db.collection(resourceType);
  return collection.findOne<T>({ id: id }, { projection: { _id: 0 } });
}
```

Then the services will get typesafety by passing in the resource interface as the generic:

``` ts
 const result = await findResourceById<fhir4.Library>(args.id, 'Library');
``` 

`result` will be of type `fhir4.Library | null`.

This is a fine compromise for now. Maybe we'll figure out a better solution later